com.notepadqq.Notepadqq.yaml

```diff
@@ -0,0 +1,43 @@
+app-id: com.notepadqq.Notepadqq
+base: io.qt.qtwebkit.BaseApp
+base-version: '5.11'
+runtime: org.kde.Platform
+runtime-version: '5.11'
+sdk: org.kde.Sdk
+command: notepadqq
+rename-icon: notepadqq
+rename-desktop-file: notepadqq.desktop
+rename-appdata-file: notepadqq.appdata.xml
+copy-icon: true
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  # Wayland access
+  - --socket=wayland
+  # Filesystem access
+  # for Open Folder
+  - --filesystem=host
+  # Network access
+  # Qt: Session management error: Could not open network socket
```

Comment:
Is it an actual error or just a warning message?

Reply:
This program uses QtNetwork and some optional features actually require access to the network.

Reply:
Something more specific than "some features"? The comment implies it's supposed to fix something – while it is just a warning.

Reply:
> Something more specific than "some features"? The comment implies it's supposed to fix something – while it is just a warning.

For example, to collect anonymous statistics (**disabled** by default).

See also: _Preferences_ → _General_ → _Collect and transmit anonymous statistics to improve Notepadqq_

https://github.com/notepadqq/notepadqq/blob/7e2446209f6adf38597ca64374bb1476e90a0567/src/ui/frmpreferences.ui#L110

Reply:
If they are disabled by default, I don't see a reason to give a text editor access to network at all.

Reply:
I'm afraid that you don't understand it correctly. It's a built-in feature, just not enabled by default.
When the user selects it in preferences, it should just work, without having to override permissions manually.

![notepadqq preferences](https://cdn.pbrd.co/images/HPJXERi.png)

Reply:
Agreed. Is this data collection also called out in the OARS tag?

Reply:
@nedrichards 

I'm afraid not. What's more, I do not see the right tag for this purpose.

Reply:
@barthalion 
> And by that I mean it should be reflected in appdata.

As I said, I don't see the right tag for this purpose.
The `social-info=moderate` seems to be the most suitable.

```
  <content_rating type="oars-1.0">
    <content_attribute id="social-info">moderate</content_attribute>
  </content_rating>
```

However, Steam reports here `social-info=none`:
https://github.com/flathub/com.valvesoftware.Steam/blob/master/com.valvesoftware.Steam.appdata.xml#L60

If we are going this way, it should get `mild` because of update checker and self-updater, `moderate` because of Steam hardware and software Survey, or even `intense`, because for sure it operates on sensitive data.
This classification has been approved not by a random person but by @nedrichards himself.
https://github.com/flathub/com.valvesoftware.Steam/pull/49

As you can see, it is very vague and unclear. Everyone chooses rating at their own discretion, while the biggest ones (Google, Mozilla) don't care about it at all.

Personally, I'm not a fan of forcing OARS for each software. It makes sense for games and some commercial messengers, but it doesn't suit everything.
I can discuss the addition of OARS information, but I must have a starting point. Give me a rating to consider, then we'll talk.

Reply:
Well, I think it's just wrong about Steam and as I wasn't involved in that discussion, you can't really bring it as counterpoint. `moderate` is described as `Sharing diagnostic data not identifiable to the user, e.g. profiling data` in [OARS generator](https://hughsie.github.io/oars/generate.html) so it sounds like a good choice (although I didn't check what exactly gets send by notepadqq).